### PR TITLE
Modify webpack for transpiling

### PR DESCRIPTION
**Pull request by mpaarating**
This PR is in reference to #59. The biggest changes here are in the `webpack.config.js`. There are a lot of opinion things that occur in this PR (mostly renaming things), so if no one likes the names, let me know and I'll revert them! 

There's some other things that need done such as adding a "production" build with minification etc.

Thanks for all of your patience while I found time here and there to work on it!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,65 +1,87 @@
 {
```
Comment (underscoredotspace) [inline package.json]: The white-space change in this file makes it difficult to see the functional changes. I know we discussed getting a prettier pre-commit hook for future, but maybe you could fix this one before merge?

**Comment by underscoredotspace** [comment]
I don't have bandwidth to do a good dig through this (job hunting/code tests/bootcamp graduation 😓), but at first blush it looks good. I'll come back and add more comments if I have time later though.

File: package.json (hunk 1 above)
Comment (mpaarating) [reply package.json]: I can totally do that. As a possible interum solution, could you see if the `hide whitespace changes` option in diff settings helps at all? Sorry for all the white space/formatting changes.

image

File: package.json [hunk 2]
```diff
@@ -1,65 +1,87 @@
...
+  "scripts": {
+    "start": "npm run watch",
+    "watch": "npm-run-all --parallel watch:server watch:build",
+    "watch:build": "webpack --progress --colors --watch",
+    "watch:server": "nodemon index.js",
+    "test": "jest",
+    "test:watch": "jest --watchAll"
+  },
+  "dependencies": {
+    "azure-iothub": "^1.2.5",
```
Comment (mtheoryx) [inline package.json]: We don't have to change this, but I'm a bit leery of having carrot-prefix on dependency numbers. I know that's the default when you `npm install` things, but... bad things can happen sometimes.

File: public/index.html [hunk 3]
```diff
@@ -1,27 +1,24 @@
...
+  <link href="https://fonts.googleapis.com/css?family=Chivo|Nanum+Pen+Script" rel="stylesheet">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="css/style.css">
+</head>
+
+<body>
+
+  <div id="app"></div>
+
+  <script src="js/dist/vendors~main.bundle.js"></script>
```
Comment (mtheoryx) [inline public/index.html]: Is that `~` in the filename?

File: public/js/components/BaseCard.vue [hunk 4]
```diff
@@ -32,13 +57,17 @@ export default Vue.component('card-base', {
...
-    }
+    };
   },
   methods: {
     onMouseDown: function(event) {
       const allowedModes = ['unlocked', 'demo'];
       const excludedNodes = ['INPUT', 'TEXTAREA', 'SELECT', 'LABEL'];
-      if (!this.dragging && !excludedNodes.includes(event.target.tagName) && allowedModes.includes(this.editMode)) {
+      if (
+        !this.dragging &&
```
Comment (mtheoryx) [inline public/js/components/BaseCard.vue]: Excellent, thanks for splitting this up. Great example for the future!

File: public/index.html (hunk 3 above)
Comment (mtheoryx) [reply public/index.html]: Disregard! Wasn't familiar with that notation!

**Comment by mpaarating** [comment]
So I had joked about the line chart card working on my machine, but it literally does. Could others confirm that it doesn't/does work on their machines? 

[Image from Gyazo](https://gyazo.com/fc42272d74a926647ff08fab694f504b)

File: package.json (hunk 2 above)
Comment (mpaarating) [reply package.json]: I totally agree. I'll lockdown versioning a bit better before this gets merged.

**Comment by mpaarating** [comment]
No worries! I should have suggested that earlier and it should really be part of the testing steps that a contributor provides in the PR (I should have included some sort of testing steps also). I suppose a PR template might be helpful in the future :D

File: package.json (hunk 2 above)
Comment (underscoredotspace) [reply package.json]: do we think this kind of change belongs in this PR? caret deps exist prior to this. I feel like we're good to merge.

Comment (mpaarating) [reply package.json]: Agreed, I'll put up another PR in the next day or so to address the versioning in this file.

**Pull request merged**